apps/definition_kafka/lib/kafka/topic.ex

```diff
@@ -0,0 +1,64 @@
+defmodule Kafka.Topic do
+  use Definition, schema: Kafka.Topic.V1
+
+  defstruct version: 1,
+            pid: nil,
+            endpoints: nil,
+            name: nil,
+            partitions: 1,
+            partitioner: :default,
+            key_path: []
+
+  def on_new(struct) do
+    endpoints =
+      struct.endpoints
+      |> List.wrap()
+      |> Enum.map(&fix_endpoint/1)
+
+    struct
+    |> Map.put(:endpoints, endpoints)
+    |> Map.update!(:partitioner, &fix_partitioner/1)
+    |> Ok.ok()
+  end
+
+  defp fix_partitioner(partitioner) when is_binary(partitioner) do
+    String.to_atom(partitioner)
+  end
+
+  defp fix_partitioner(partitioner), do: partitioner
+
+  defp fix_endpoint([host, port]) when is_binary(host) and is_integer(port) do
+    {String.to_atom(host), port}
+  end
+
+  defp fix_endpoint(endpoint), do: endpoint
+
+  defimpl Source do
```

Comment:
Why implement and then delegate here instead of just implementing at the source/dest module? I can see it making sense to do this as a facade / high level interface, but I could also see just grabbing the source and or destination module if I needed it.

Reply:
Because we implementing Source for the Kafka.Topic struct, so it felt natural that he imp be there.  We could move it to the the source.ex file, but the impl is still for Kafka.Topic.  My vote is to leave it where it is.